### PR TITLE
add python customer logo scraper and editor

### PR DESCRIPTION
**Pull request by Vadman97**
## Summary

Add automated scraper for customer logos that edits logos for a dark background.

## How did you test this change?

Running script locally

## Are there any deployment considerations?

no

## Does this work require review from our design team?

no

### CODE REVIEW

File: scripts/scrape_icons.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            yield domain, r.content
+        except (AssertionError, requests.RequestException) as e:
+            print(f"failed to fetch domain {domain} - {e}")
+            continue
+
+
+def get_domain_logos(content: str):
+    soup = BeautifulSoup(content, "html.parser")
+    for elem in soup.find_all():
+        url = elem.attrs.get("src") or elem.attrs.get("href")
```
Comment (ccschmitz) [inline scripts/scrape_icons.py:23]: When would the image be in the `href`?

File: scripts/scrape_icons.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            assert r.ok, r.status_code
+            yield domain, r.content
+        except (AssertionError, requests.RequestException) as e:
+            print(f"failed to fetch domain {domain} - {e}")
+            continue
+
+
+def get_domain_logos(content: str):
+    soup = BeautifulSoup(content, "html.parser")
+    for elem in soup.find_all():
```
Comment (ccschmitz) [inline scripts/scrape_icons.py:22]: Think we should limit to `img` tags?


```suggestion
    for elem in soup.find_all("img"):
```

File: scripts/scrape_icons.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+def edit_logo(path: str):
+    file_name = os.path.basename(path)
+    out_name = f"{file_name}.png"
+    call = subprocess.check_call(
+        [
+            "gimp",
+            "-i",
+            "-b",
+            f'(process-logo "{file_name}" "{out_name}")',
```
Comment (ccschmitz) [inline scripts/scrape_icons.py:80]: Just curious, where does this command come from?

**Review by ccschmitz — APPROVED** [review]
Looks cool! Just curious, what is this being used for?

File: scripts/scrape_icons.py (hunk 2 above)
Comment (Vadman97) [reply scripts/scrape_icons.py:23]: wanted to also check `meta favicon` tags, but i think you're right - all of what i found was from `img` tags

**Pull request merged**